components/common/src/templating/config.rs

```diff
@@ -644,24 +644,27 @@ fn ensure_directory_structure(root: &Path, file: &Path, user: &str, group: &str)
     let dir = file.parent().unwrap();
 
     if !dir.exists() {
-        std::fs::create_dir_all(&dir)?;
+        std::fs::create_dir_all(&dir).map_err(Error::IO)?;
         for anc in dir.ancestors().take_while(|&d| d != root) {
-            set_permissions(&anc, &user, &group)?;
+            set_permissions(&anc, &user, &group).map_err(Error::HabitatCore)?;
```

Comment:
It's not clear to me why these `map_err`s are added. The code builds without them. Is there somewhere else we handle those specifically mapped errors?